src/pydicom/dataset.py

```diff
@@ -518,6 +518,51 @@ def add_new(self, tag: TagType, VR: str, value: Any) -> None:
         """
         self.add(DataElement(tag, VR, value))
 
+    def add_private(
```

Comment:
I'm not sure about the name. `add_private_item` would be complementary to `get_private_item`, but we are not passing an item here. `add_new_private` and `add_private_new` to match `add_new` both sound strange. Any idea?

Reply:
Hmm.. I like `add_new_private` as a nice parallel to the existing `add_new`.  

I thought briefly about whether it just be added to `add_new` with an extra parameter for the private creator. But it would need some extra logic: error if private creator specified but is not private tag, or if private tag but no creator specified.  And worse, because a full tag is passed rather than element offset, a lot more logic to check that it is using the correct private group if previously created.  I think it is less confusing to keep it as a separate method, and then easier to point to in documentation as well.

Reply:
> I thought briefly about whether it just be added to add_new with an extra parameter for the private creator. 

Yes, I had the same thought, and dopped it for the same reasons.